sdk/core/src/services/purl2cpe/service.rs

```diff
@@ -0,0 +1,486 @@
+                // If the database is erroring out, then we can't go on.
+                Err(err) => {
+                    return error_result(
+                        format!("==> unable to get cpe, database error: {}", err).as_str(),
+                    )
+                }
+
+                // The Database gave us a response
+                Ok(option) => match option {
+                    // The response was empty
+                    None => {
+                        println!(
+                            "==> unable to get cpe for purl({}), database response is empty",
+                            full_purl
+                        );
+
+                        Ok(None)
+                    }
+
+                    // the response had CPEs!
+                    Some(purl_2_cpes) => {
+                        let cpes: Vec<String> = purl_2_cpes.cpes;
+
+                        // If there is only one cpe, then return it.
+                        if cpes.len() == 1 {
+                            let cpe = cpes.get(0).unwrap().to_owned();
+                            return Ok(Some(cpe));
+                        }
+
+                        Ok(find_cpe(String::from("*"), cpes))
+                    }
+                },
+            }
+        };
+    }
+
+    /// Clones the purl2cpe repository at:
+    ///     https://github.com/scanoss/purl2cpe
+    pub fn clone_purl2cpe_repo(&self) -> Result<String, Error> {
+        if Path::new(CLONE_PATH).is_dir() {
+            self.remove_purl2cpe_clone()?;
+        }
+
+        let token =
+            github_pat().map_err(|err| Error::Purl2Cpe(format!("Unable to get pat: {}", err)))?;
+        Git::new(String::from(PURL_2_CPE_URL))
+            .clone_repo(CLONE_PATH, Some(token))
+            .map_err(Error::Git)?;
+
+        Ok(String::from(CLONE_PATH))
+    }
+
+    /// Method to find yaml files
+    pub fn find_purl_yaml_files(&self) -> Result<Vec<String>, Error> {
+        let files = Git::new(String::from(PURL_2_CPE_URL))
+            .find(String::from("purls.yml"), String::from(CLONE_PATH))
+            .map_err(Error::Git)?;
+
+        Ok(files)
+    }
+
+    /// Removes a cloned repository from the filesystem.
+    pub fn remove_purl2cpe_clone(&self) -> Result<(), Error> {
+        Git::remove_clone(CLONE_PATH).map_err(Error::Git)
+    }
+}
+
+/// Represents all exposed Errors for this task.
+#[derive(Error, Debug)]
+pub enum Error {
+    /// Error to handle percent encoding errors
+    #[error(transparent)]
+    PercentEncoding(#[from] FromUtf8Error),
+    /// Error to map from platform Git Error
+    #[error(transparent)]
+    Git(#[from] platform::git::error::Error),
+    /// Error to map from platform, specifically Mongo
+    #[error(transparent)]
+    Mongo(#[from] platform::Error),
+    /// Error for Perl2Cpe general error
+    #[error("purl2cpe error: {0}")]
+    Purl2Cpe(String),
+}
+
+/*
+ * Why are so many of these tests "debug manual only"?
```

Comment:
Because we didn't have the Harbor Test Org.  We do now, so I hope to turning a lot of these on as we run local and CI tests against that org.  Stay tuned.

Reply:
Will do.  We could use better tests